### ISSUE TITLE
IE11 crashes when source is loaded before Promises polyfill

### ISSUE THREAD

**Issue by aitorllj93** [limonte/sweetalert2#767]
Hi,

I've seen that in your latest release you changed the isDefined checking from this:

```js
if (typeof Promise === 'undefined') {
```
to this:

```js
if (!isDefined(Promise)) {
```

That line of code crashes on IE11 and stops the file being loaded, forcing us to always load the Promise polyfill before. 

My source code is bundled together in one single file and it has a method that loads polyfills on demand after checking the navigator features. Now, I can't do that with Promises due to the crash I've reported.

I would like to know if this is ok for you, or have plans on adressing this issue, because if not, I need to change some things in my code (I think I will end adding the Promise polyfill to the source bundle before your source is added to it)

Thanks for your response

**Comment by toverux** [comment]
@limonte Hm, I'm surprised that we use a function here because simply referencing `Promise` alone will make all runtimes crash if it's not defined, hence the `typeof` check which is a language construct that permits to address undefined symbols safely.

See:

![image](https://user-images.githubusercontent.com/1343250/33992935-c6cbe192-e0d4-11e7-8f93-5a354122a4bd.png)

**Comment by limonte** [comment]
My bad :man_facepalming: will fix asap

**Comment by limonte** [comment]
Fixed in v7.0.10, thanks @aitorllj93 for reporting and @toverux for explaining my fuckup
